packages/pigeon/lib/ast.dart

```diff
@@ -0,0 +1,77 @@
+/// Enum that represents where an [Api] is located, on the host or Flutter.
+enum ApiLocation {
+  /// The API is for calling functions defined on the host.
+  host,
+
+  /// The API is for calling functions defined in Flutter.
+  flutter,
+}
+
+/// Superclass for all AST nodes.
+class Node {}
+
+/// Represents a method on an [Api].
+class Func extends Node {
```

Comment:
`Function` is a reserved type in Dart.  I promise I'll change it to Method in a follow up PR.  Renaming it won't affect regular usage.

Reply:
Ok renamed it.  My original resistance to it was I was moving so fast I was going to get to it shortly.  Since the review process is taking some time I can change it here.